File: java/Product.java
public class Product extends Shirt {
    private int idProduct;
    private String name;
    private String brand;
    private int price;

    // Default constructor
    public Product() {
        super(); // Calling superclass constructor
        this.idProduct = 0;
        this.name = "";
        this.brand = "";
        this.price = 0;
    }

    // Parameterized constructor
    public Product(String size, String material, String gender, String color, String sleeveType, int idProduct, String name, String brand, int price) {
        super(size, material, gender, color, sleeveType); // Calling superclass constructor
        this.idProduct = idProduct;
        this.name = name;
        this.brand = brand;
        this.price = price;
    }

    // Getters
    public int getIdProduct() {
        return idProduct;
    }

    public String getName() {
        return name;
    }

    public String getBrand() {
        return brand;
    }

    public int getPrice() {
        return price;
    }

    // Setters
    public void setIdProduct(int idProduct) {
        this.idProduct = idProduct;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setBrand(String brand) {
        this.brand = brand;
    }

    public void setPrice(int price) {
        this.price = price;
    }
}

File: cpp/Product.cpp
#include "Product.hpp"

// Default constructor
Product::Product() : Shirt(), idProduct(0), name(""), brand(""), price(0) {}

// Parameterized constructor
Product::Product(std::string size, std::string material, std::string gender, std::string color, std::string sleeveType,
                 int idProduct, std::string name, std::string brand, int price)
    : Shirt(size, material, gender, color, sleeveType), idProduct(idProduct), name(name), brand(brand), price(price) {}

// Getters
int Product::getIdProduct() {
    return idProduct;
}

std::string Product::getName() {
    return name;
}

std::string Product::getBrand() {
    return brand;
}

int Product::getPrice() {
    return price;
}

// Setters
void Product::setIdProduct(int idProduct) {
    this->idProduct = idProduct;
}

void Product::setName(std::string name) {
    this->name = name;
}

void Product::setBrand(std::string brand) {
    this->brand = brand;
}

void Product::setPrice(int price) {
    this->price = price;
}
